src/WorkspaceDependencyViewExtension/WorkspaceDependencyViewExtension.cs

```diff
@@ -229,6 +233,30 @@ internal void DependencyRegen(WorkspaceModel ws, bool forceCompute = false)
                         {
                             packageDependencyInfo.Path = targetInfo.RootDirectory;
                         }
+                        if (!PythonPackageVersionMismatchWarningDismissed)
+                        {
+                            if (packageDependencyInfo.State == PackageDependencyState.IncorrectVersion)
+                            {
+                                //check if a version mismatch for a python node engine has been detected
+                                var pyNodes = packageDependencyInfo.Nodes.Where(x => GetNodeFromGUID(ws,x) is PythonNodeBase);
```

Comment:
I think we can clean this up a bit. We get all the python nodes, then only use the first and end up casting it twice. Maybe:
`packageDependencyInfo.Nodes.Select(x => GetNodeFromGUID(ws, x)).OfType<PythonNodeBase>().FirstOrDefault()`?